### PR TITLE
security_page

### PR DESCRIPTION
**Pull request by barakshani**

### CODE REVIEW

File: docs/consensus/security.md [hunk 1]
```diff
@@ -13,4 +13,75 @@ keywords:
 - Constraining Simulation
 - Handling Equivocation
 - Detecting Long-Range Attacks -->
-For detailed security analysis please refer to our paper [Dilithium: A Proof-of-Archival-Storage Consensus Protocol for Subspace](https://github.com/subspace/consensus-v2-research-paper)
\ No newline at end of file
+
+A complex system like the Subspace protocol has many potential attack vectors, some are more general, blockchain related, 
+while other are focused on proof of capacity and specifically proof of archive storage, as used in Subspace.  
+
+This page attempts to give an overview of how such attacks are mitigated in the Subspace protocol. There are many things
```
Comment (dariolina) [inline docs/consensus/security.md]: I'd say simply "gives an overview"

Comment (barakshani) [reply docs/consensus/security.md]: Too apologetic?
Basically I'm trying to pass the message that the page should not be considered complete or comprehensive.

File: docs/consensus/security.md [hunk 2]
```diff
@@ -13,4 +13,75 @@ keywords:
 - Constraining Simulation
 - Handling Equivocation
 - Detecting Long-Range Attacks -->
-For detailed security analysis please refer to our paper [Dilithium: A Proof-of-Archival-Storage Consensus Protocol for Subspace](https://github.com/subspace/consensus-v2-research-paper)
\ No newline at end of file
+
+A complex system like the Subspace protocol has many potential attack vectors, some are more general, blockchain related, 
+while other are focused on proof of capacity and specifically proof of archive storage, as used in Subspace.
```
Comment (0xjeremyfrank) [inline docs/consensus/security.md]:
```suggestion
while other are focused on proof of capacity and specifically proof of archival storage, as used in Subspace.  
```

File: docs/consensus/security.md (hunk 1 above)
Comment (0xjeremyfrank) [reply docs/consensus/security.md]: You state that message clearly in the next sentence, so I think "gives an overview" is sufficient.

File: docs/consensus/security.md [hunk 3]
```diff
@@ -13,4 +13,75 @@ keywords:
...
 - Detecting Long-Range Attacks -->
-For detailed security analysis please refer to our paper [Dilithium: A Proof-of-Archival-Storage Consensus Protocol for Subspace](https://github.com/subspace/consensus-v2-research-paper)
\ No newline at end of file
+
+A complex system like the Subspace protocol has many potential attack vectors, some are more general, blockchain related, 
+while other are focused on proof of capacity and specifically proof of archive storage, as used in Subspace.  
+
+This page attempts to give an overview of how such attacks are mitigated in the Subspace protocol. There are many things
+ to cover, so the reader should be aware that at the current state this page is not exhaustive. Furthermore, the 
+presentation mostly highlights the techniques used.
```
Comment (0xjeremyfrank) [inline docs/consensus/security.md]: I'm not sure what this sentence is trying to say "Furthermore, the 
presentation mostly highlights the techniques used." It feels incomplete.

File: docs/consensus/security.md [hunk 4]
```diff
@@ -13,4 +13,75 @@ keywords:
...
+For detailed security analysis please refer to our paper [Dilithium: A Proof-of-Archival-Storage Consensus Protocol for Subspace](https://github.com/subspace/consensus-v2-research-paper).
+
+# Security against general blockchain attacks
+
+## Grinding on block challenges
+
+To prevent grinding on block challenges, we use the proof-of-time outputs to draw unique challenges.
+
+Unlike in proof-of-work-based blockchains, where the "challenge" for block creation comes from the previous block, in 
+Subspace this is not the case, as changing the block content does not affect the proof of space validity. Instead,
```
Comment (0xjeremyfrank) [inline docs/consensus/security.md]: The sentence structure is awkward, maybe something like:

In Subspace, unlike in proof-of-work-based blockchains where the "challenge" for block creation comes from the previous block, changing the block content does not affect the proof of space validity.

File: docs/consensus/security.md [hunk 5]
```diff
@@ -13,4 +13,75 @@ keywords:
...
+
+## Time-Memory algorithms (plot compression)
+
+We adopt the function described in the work ["Beyond Hellman's Time-Memory Trade-Offs with Applications to Proofs of Space"](https://eprint.iacr.org/2017/893) 
+as our masking function. This function is designed such that the gain in trading computation (time) over storage (memory) 
+is very small.
+
+## On-the-fly plot creation
+
+Preventing farmers from creating plots on the fly, after seeing the challenge, has two favours.
```
Comment (0xjeremyfrank) [inline docs/consensus/security.md]: Did you mean "flavors" here?

**Review by 0xjeremyfrank — COMMENTED** [review]
Very succinct and informative! Just a few comments.

File: docs/consensus/security.md (hunk 5 above)
Comment (barakshani) [reply docs/consensus/security.md]: Indeed, thanks. I guess it should all be in US English spelling?

File: docs/consensus/security.md (hunk 1 above)
Comment (barakshani) [reply docs/consensus/security.md]: I'll revise. Just to note that we should all aim for this page to be complete at some point.

File: docs/consensus/security.md (hunk 3 above)
Comment (barakshani) [reply docs/consensus/security.md]: Trying to say that detailed explanation (furthermore, security proofs) are omitted. I thought the following sentence makes this clear, but I'll revise.

File: docs/consensus/security.md (hunk 5 above)
Comment (0xjeremyfrank) [reply docs/consensus/security.md]: Probably should be US English for consistency. This one I was more pointing out it was the wrong word.

File: docs/consensus/security.md (hunk 4 above)
Comment (barakshani) [reply docs/consensus/security.md]: Better now? I had some issues with adopting your suggestion.

**Review by 0xjeremyfrank — APPROVED** [review]
LGTM

**Review by dariolina — APPROVED** [review]

**Pull request merged**